### PR TITLE
Geojsonlayer improvements

### PR DESCRIPTION
**Pull request by abusquets**

### CODE REVIEW

File: layerserver/views.py [hunk 1]
```diff
@@ -120,7 +120,12 @@ def dispatch(self, request, *args, **kwargs):
         self.pagination_class = self.get_pagination_class(self.layer)
         self.filter_fields = []
         self._fields = {}
+        only_fields = self.request.GET.get('fields', None)
```
Comment (manelclos) [inline layerserver/views.py]: maybe have [] as de default value instead of None to simplify the code?

File: layerserver/views.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def GeoJSONLayerView(request, layer_name):
     layer = GeoJsonLayer.objects.filter(
         active=True,
         name=layer_name).first()
-    if layer.visibility == 'private' and not request.user.is_authenticated:
+    if not layer or (layer.visibility == 'private' and not request.user.is_authenticated):
```
Comment (manelclos) [inline layerserver/views.py]: "not layer" should produce a 404, and checking private layers without being authenticated too.

File: layerserver/utils.py [hunk 3]
```diff
@@ -46,26 +49,32 @@ def geojsonlayer_refresh(pk):
...
         else:
-            content = ContentFile(r.text)
-            if content:
-                remote_file = os.path.join(
-                    settings.MEDIA_ROOT, layer.service_path, 'remote.json')
-                if os.path.exists(remote_file):
-                    os.remove(remote_file)
-                layer.data_file.save('remote.json', content, save=True)
-                layer.last_fetch_on = timezone.localtime()
+            if str(r.status_code).startswith('2'):
```
Comment (manelclos) [inline layerserver/utils.py]: convert to int, check >= 200 < 300

File: layerserver/utils.py [hunk 4]
```diff
@@ -46,26 +49,32 @@ def geojsonlayer_refresh(pk):
 
 def geojsonlayer_refresh_layer(layer):
     if layer.url:
+        headers = {}
+        if layer.headers:
+            matches = re.findall(r'^([^=#\n\r][^=]*)=(.*)$', layer.headers, flags=re.M)
```
Comment (manelclos) [inline layerserver/utils.py]: please explain regular expression purpose

File: layerserver/views.py (hunk 1 above)
Comment (abusquets) [reply layerserver/views.py]: I need to know if there is a fields parameter.

File: layerserver/utils.py (hunk 3 above)
Comment (abusquets) [reply layerserver/utils.py]: Ok

File: layerserver/utils.py (hunk 4 above)
Comment (abusquets) [reply layerserver/utils.py]: ok

File: layerserver/views.py (hunk 2 above)
Comment (abusquets) [reply layerserver/views.py]: You are right :+1:

**Pull request merged**